discovery_engine_core/core/src/engine.rs

```diff
@@ -1615,9 +1623,9 @@ async fn update_stacks<'a>(
         }
     }
 
-    // only return an error if all stacks that were ready to get new items failed
-    if !errors.is_empty() && errors.len() >= ready_stacks {
-        Err(Error::Errors(errors))
+    // only return an error if all stacks (including exploration) failed
+    if errors.len() > stacks.len() {
```

Comment:
:art:??

This line is quite easy to misunderstand/confuse.

I'm wondering if following would be better but I'm really not sure about it.

```suggestion
    if errors.len() >= stacks.len() + 1 {
```

Reply:
i implemented it before like you suggested, but then clippy complained. i'll change it back and ignore clippy for this one.